print-medsa.c

```diff
@@ -190,3 +202,59 @@ medsa_print(netdissect_options *ndo,
 trunc:
 	nd_print_trunc(ndo);
 }
+
+/* The DSA tag is 4 bytes. It has the same content as the last 4 bytes
+ * as the EDSA tag.
+ */
+
+u_int
+mdsa_tag_if_print(netdissect_options *ndo, const struct pcap_pkthdr *h,
+		  const u_char *bp)
+{
+	const struct mdsa_pkthdr *mdsa;
+	const struct ether_header *ehp;
+	int old_eflag = ndo->ndo_eflag;
+	u_int caplen = h->caplen;
+	u_int length = h->len;
+	int ret;
+
+	ndo->ndo_protocol = "mdsa";
+	if (caplen < ETHER_SA_OFFSET + MDSA_TAG_LEN) {
+		nd_print_trunc(ndo);
+		return (caplen);
+	}
+
+	if (length < ETHER_SA_OFFSET + MDSA_TAG_LEN) {
+		nd_print_trunc(ndo);
+		return (length);
+	}
+
+	ehp = (const struct ether_header *)bp;
+	if (ndo->ndo_eflag)
+		ND_PRINT("%s > %s, ",
+			     etheraddr_string(ndo, ehp->ether_shost),
+			     etheraddr_string(ndo, ehp->ether_dhost));
+
+	mdsa = (const struct mdsa_pkthdr *)(bp + ETHER_SA_OFFSET);
+
+	if (!ndo->ndo_eflag)
+		ND_PRINT("MDSA %u.%u:%u: ",
+			  SRC_DEV(mdsa), SRC_PORT(mdsa), VID(mdsa));
+	else
+		mdsa_print_full(ndo, mdsa, caplen - ETHER_SA_OFFSET);
+
+	/* We printed the Ethernet header already */
+	ndo->ndo_eflag = 0;
+
+	/* Parse the Ethernet frame regularly telling how big the non
+	 * standard Ethernet header is.
+	 *
+	 * +-----------++-----------++----------------++--------------+
+	 * | MAC DA (6)|| MAC SA (6)||Broadcom tag (4)||Type/Length(2)|
```

Comment:
Presumably you mean "Marvell tag" or "DSA tag" here, rather than "Broadcom tag"?

Reply:
Yep. I Cut/paste from Florians' broadcom code, and did not fix it up.